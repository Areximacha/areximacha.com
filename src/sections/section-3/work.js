import React from 'react'

import WorkSection from './work.style'

const Work = () => (
  <WorkSection id='work'>
    <h1>work</h1>
  </WorkSection>
)

export default Work
