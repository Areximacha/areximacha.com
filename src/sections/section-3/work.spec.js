import React from 'react'
import { render } from 'react-testing-library'
import renderer from 'react-test-renderer'

import { PureWork as Work } from './work'

describe('<Work/>', () => {
  const data = {
    allMarkdownRemark: {
      edges: [
        {
          node: {
            html: `test1`,
            frontmatter: {
              title: `arcadia`,
            },
          },
        },
        {
          node: {
            html: `test2`,
            frontmatter: {
              title: `batman`,
            },
          },
        },
      ],
    },
  }

  describe('@render', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<Work data={data} />).toJSON()

      expect(tree).toMatchSnapshot()
    })

    it('renders buttons by incoming data', () => {
      const { queryByTestId } = render(<Work data={data} />)

      data.allMarkdownRemark.edges.map(({ node: { frontmatter: { title } } }) =>
        expect(queryByTestId(`test-${title}`)).toBeTruthy()
      )
    })

    it('renders the default node and not any others on initial render', () => {
      const { queryByTestId } = render(<Work data={data} />)

      expect(queryByTestId('test-description')).toHaveTextContent('test1')
      expect(queryByTestId('test-description')).not.toHaveTextContent('test2')
    })
  })
})
