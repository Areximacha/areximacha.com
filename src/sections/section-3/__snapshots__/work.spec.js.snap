// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Work/> @render renders correctly 1`] = `
<section
  className="css-7qcwkc"
  id="work"
>
  <div
    className="arcadia css-nokm5q"
  />
  <div
    className="css-bbgkix"
  >
    <div
      className="css-xq4zy8"
    >
      <h3>
        Selected Projects
      </h3>
      <div
        className="css-kc7c0l"
      >
        <button
          className="work-arcadia active"
          data-testid="test-arcadia"
          onClick={[Function]}
          type="button"
        />
        <button
          className="work-batman"
          data-testid="test-batman"
          onClick={[Function]}
          type="button"
        />
      </div>
    </div>
    <div
      className="css-f9b1t7"
      dangerouslySetInnerHTML={
        Object {
          "__html": "test1",
        }
      }
      data-testid="test-description"
    />
  </div>
</section>
`;
